tests/checks-nix-gc.nix

```
@@ -14,6 +14,6 @@ in
 
     echo checking nix-gc service in /Library/LaunchDaemons >&2
     grep "<string>org.nixos.nix-gc</string>" ${config.out}/Library/LaunchDaemons/org.nixos.nix-gc.plist
-    ! grep "<key>UserName</key>" ${config.out}/Library/LaunchDaemons/org.nixos.nix-gc.plist
+    ! grep "<key>UserName</key>" ${config.out}/Library/LaunchDaemons/org.nixos.nix-gc.plist || false
```

```suggestion
    grep "<key>UserName</key>" ${config.out}/Library/LaunchDaemons/org.nixos.nix-gc.plist || false
```

I'm not sure what the order of operations is but it seems clearer without using `!`

---

Not the same; it would have to be `&& exit 1` which is longer (albeit possibly clearer). Another option would be to use subshells.

---

I think using subshells would be clearer than `&& exit 1`

---

Sure; force-pushed. My feeling was that they're all confusing so I kept the one that's more efficient and kept the `!` at the start, but I'm happy to defer here :)

There already a couple `&& return`s in the tests but I didn't touch them since they work as-is. (Although the error is that they failed to create `out` rather than an explicit error code, which is unfortunate.)